### ISSUE TITLE
Unexpected Error

### ISSUE THREAD

**Issue by ChrisSargent** [csstools/stylelint-prefer-nesting#7]
Hey there!

Given the following sass code:

```scss
.wp__post {
  > .wp__core-paragraph {
    outline: 1px solid green;
  }

  .wp__core-paragraph {
    outline: 1px solid red;
  }
}
```

I am getting the following error: `Expected "> .wp__core-paragraph" inside ".wp__core-paragraph".   csstools/use-nesting`

I can't think of a way to rewrite this rule to satisfy this error so I wonder if it's a bug?

**Comment by jonathantneal** [comment]
Hi @ChrisSargent, this plugin enforces CSS Nesting (https://drafts.csswg.org/css-nesting-1/) which is different than Sass nesting. The `&` is needed in CSS nesting. However, should this plugin still suite your needs, the `&` is also compatible with Sass, which might make your code more _future-proof_, so to speak. I hope that helps, and is not too much of a disappointment.

**Comment by ChrisSargent** [comment]
I should say, I am using SCSS. Okay, so the equivalent (latest) CSS code should be:

```css
.parent {
  .child {
    outline: 1px solid red;
       @nest > & {
      outline: 1px solid orange;
    }
  }
}
```
Using https://jonneal.dev/precss/ translates to...

```css
.parent .child {
    outline: 1px solid red
  }
.parent > .child {
    outline: 1px solid orange;
}
```
Which is as expected...

However, working in scss land again, using `@nest` doesn't seem supported - the same code as above just results in this being output:

```scss
@nest > & {
  .parent .child {
    outline: 1px solid orange;
  }
}
```
which is wrong.

Trying to nest in scss without using `@nest`...

Input:
```scss
.parent {
  .child {
    outline: 1px solid red;
    & > {
      outline: 1px solid orange;
    }
  }
}
```
Output:
```css
.parent .child {
  outline: 1px solid red;
}

.parent .child > {
    outline: 1px solid orange;
}
```

Input:
```scss
.parent {
  .child {
    outline: 1px solid red;
    > & {
      outline: 1px solid orange;
    }
  }
}
```

Output:
```css
.parent .child {
  outline: 1px solid red;
}
> .parent .child {
    outline: 1px solid orange;
}
```

Input:
```scss
.parent {
  .child {
    outline: 1px solid red;
    & > & {
      outline: 1px solid orange;
    }
  }
}
```

Output:
```css
.parent .child {
  outline: 1px solid red;
}

.parent .child > .parent .child {
    outline: 1px solid orange;
}
```

Basically, I don't think it's possible to satisfy this when using scss, unless I'm missing some Sass trick... hmmm.

p.s. I'm not saying there's something wrong with your plugin, but I'm adding the code above as a reference as much as anything else, I'm not quite sure how this can be worked around.

**Comment by romainmenke** [comment]
A lot of time has passed since this issue was opened and the nesting spec has radically changed.

Going to close this in favor of a more in depth review of this plugin
